data-prepper-api/src/main/java/org/opensearch/dataprepper/model/plugin/ExtensionPlugin.java

```diff
@@ -20,4 +20,9 @@ public interface ExtensionPlugin {
      * @param extensionPoints The {@link ExtensionPoints} wherein the extension can extend behaviors.
      */
     void apply(ExtensionPoints extensionPoints);
+
+    /**
+     * Close resources used by the extension.
+     */
+    void shutdown();
```

Comment:
Let's make this a default method so that sub-classes don't have to implement with an empty implementation. Then you can remove the empty implementations that you had in this PR.